### PR TITLE
Use SIP grants

### PR DESCRIPTION
**Pull request by dennwc**

### CODE REVIEW

File: auth.go [hunk 1]
```diff
@@ -27,9 +27,12 @@ type authBase struct {
 	apiSecret string
 }
 
-func (b authBase) withAuth(ctx context.Context, grant auth.VideoGrant) (context.Context, error) {
+func (b authBase) withAuth(ctx context.Context, grant auth.VideoGrant, sip *auth.SIPGrant) (context.Context, error) {
```
Comment (davidzhao) [inline auth.go:30]: I think it'd be better to have two separate functions for these things, since we very rarely use these together

Comment (davidzhao) [reply auth.go:30]: i.e. `withSIPAuth`

Comment (dennwc) [reply auth.go]: Updated, kept a single method, but used interface type as argument. Looks a lot nicer. WDYT?

Should probably be moved to `protocol/auth` eventually.

**Review by davidzhao — APPROVED** [review]
nice, even better!

**Pull request merged**